paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentConfirmationDefinition.kt

```diff
@@ -0,0 +1,56 @@
+package com.stripe.android.paymentsheet
+
+import android.os.Parcelable
+import androidx.activity.result.ActivityResultCaller
+import com.stripe.android.core.strings.ResolvableString
+import com.stripe.android.model.StripeIntent
+
+internal interface PaymentConfirmationDefinition<
+    TConfirmationOption : PaymentConfirmationOption,
+    TLauncher,
+    TLauncherArgs,
+    TLauncherResult : Parcelable
+    > {
+    suspend fun action(
+        confirmationOption: TConfirmationOption,
+        intent: StripeIntent,
+    ): ConfirmationAction<TLauncherArgs>
+
+    fun launch(
+        launcher: TLauncher,
+        arguments: TLauncherArgs,
+        confirmationOption: TConfirmationOption,
+        intent: StripeIntent,
+    )
```

Comment:
I chose to separate launching from the actual performed action as in between figuring out what to do and needing to launch, we need to handle process death as well as storing the deferred confirmation type. In the future, we also will need to update to update the `Preconfirming` state to notify when we're launching into a pre-confirmation flow.